go/tasks/plugins/k8s/sagemaker/builtin_training.go

```diff
@@ -0,0 +1,241 @@
+	if role == "" {
+		role = cfg.RoleArn
+	}
+
+	trainingJob := &trainingjobv1.TrainingJob{
+		Spec: trainingjobv1.TrainingJobSpec{
+			AlgorithmSpecification: &commonv1.AlgorithmSpecification{
+				// If the specify a value for this AlgorithmName parameter, the user can't specify a value for TrainingImage.
+				// in this Flyte plugin, we always use the algorithm name and version the user provides via Flytekit to map to an image
+				// so we intentionally leave this field nil
+				AlgorithmName:     nil,
+				TrainingImage:     ToStringPtr(trainingImageStr),
+				TrainingInputMode: commonv1.TrainingInputMode(inputModeString),
+				MetricDefinitions: metricDefinitions,
+			},
+			// The support of spot training will come in a later version
+			EnableManagedSpotTraining: nil,
+			HyperParameters:           staticHyperparams,
+			InputDataConfig: []commonv1.Channel{
+				{
+					ChannelName: ToStringPtr(TrainPredefinedInputVariable),
+					DataSource: &commonv1.DataSource{
+						S3DataSource: &commonv1.S3DataSource{
+							S3DataType: "S3Prefix",
+							S3Uri:      ToStringPtr(trainPathLiteral.GetScalar().GetBlob().GetUri()),
+						},
+					},
+					ContentType: ToStringPtr(apiContentType),
+					InputMode:   inputModeString,
+				},
+				{
+					ChannelName: ToStringPtr(ValidationPredefinedInputVariable),
+					DataSource: &commonv1.DataSource{
+						S3DataSource: &commonv1.S3DataSource{
+							S3DataType: "S3Prefix",
+							S3Uri:      ToStringPtr(validationPathLiteral.GetScalar().GetBlob().GetUri()),
+						},
+					},
+					ContentType: ToStringPtr(apiContentType),
+					InputMode:   inputModeString,
+				},
+			},
+			OutputDataConfig: &commonv1.OutputDataConfig{
+				S3OutputPath: ToStringPtr(outputPath),
+			},
+			CheckpointConfig: nil,
+			ResourceConfig: &commonv1.ResourceConfig{
+				InstanceType:   sagemakerTrainingJob.GetTrainingJobResourceConfig().GetInstanceType(),
+				InstanceCount:  ToInt64Ptr(sagemakerTrainingJob.GetTrainingJobResourceConfig().GetInstanceCount()),
+				VolumeSizeInGB: ToInt64Ptr(sagemakerTrainingJob.GetTrainingJobResourceConfig().GetVolumeSizeInGb()),
+				VolumeKmsKeyId: ToStringPtr(""), // TODO: Not yet supported. Need to add to proto and flytekit in the future
+			},
+			RoleArn: ToStringPtr(role),
+			Region:  ToStringPtr(cfg.Region),
+			StoppingCondition: &commonv1.StoppingCondition{
+				MaxRuntimeInSeconds:  ToInt64Ptr(86400), // TODO: decide how to coordinate this and Flyte's timeout
+				MaxWaitTimeInSeconds: nil,               // TODO: decide how to coordinate this and Flyte's timeout and queueing budget
+			},
+			TensorBoardOutputConfig: nil,
+			Tags:                    nil,
+			TrainingJobName:         &jobName,
+		},
+	}
+	logger.Infof(ctx, "Successfully built a training job resource for task [%v]", taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName())
+	return trainingJob, nil
+}
+
+func (m awsSagemakerPlugin) getTaskPhaseForTrainingJob(
```

Comment:
is this different for every job type? or are the jobSTatus common?

Reply:
The output handling is different for different type of tasks

Reply:
the job statuses are not common either

Reply:
Ya I realized once I went through all of them